lib/core/src/Cardano/Wallet/Orphans.hs

```diff
@@ -65,3 +74,18 @@ instance NFData TxMetadataValue where
 -- Defined here so that other types with use PastHorizonException can have Eq.
 instance Eq PastHorizonException where
     a == b = displayException a == displayException b
+
+instance ToText NominalDiffTime where
+    toText = T.pack . show
+
+instance FromText NominalDiffTime where
+    fromText t = case T.splitOn "s" t of
+        [v,""] -> maybe (Left err) (Right . toDiffTime) (readMaybe $ T.unpack v)
+        _ -> Left err
+      where
+        toDiffTime = fromIntegral :: Natural -> NominalDiffTime
+        err = TextDecodingError $ unwords
+            [ "Cannot parse given time duration."
+            , "Values must be given as whole positive seconds,"
+            , "and must finish with \"s\". For example: \"3s\", \"3600s\", \"42s\"."
+            ]
```

Comment:
Alternatively, could be declared in the `text-class` package directly to avoid the orphan instance?

Reply:
Yeah.